src/Evolution/Systems/ScalarWave/BoundaryCorrections/UpwindPenalty.hpp

```diff
@@ -0,0 +1,279 @@
+ * the same footing.
+ *
+ * For the scalar wave system the correction is:
+ *
+ * \f{align}{
+ *   D_{\Psi} &= \tilde{\lambda}_{v^{\Psi}}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},\Psi}
+ *                - \tilde{\lambda}_{v^{\Psi}}^{\mathrm{int}}
+ *                v^{\mathrm{int},\Psi}, \\
+ *   D_{\Pi} &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *             v^{\mathrm{ext},+} +
+ *             \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *             v^{\mathrm{ext},-}\right)
+ *             + \tilde{\lambda}_{v^\Psi}^\mathrm{ext}\gamma_2
+ *             v^{\mathrm{ext},\Psi}
+ *             \notag \\
+ *           &-\frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *             v^{\mathrm{int},+} +
+ *             \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *             v^{\mathrm{int},-}\right)
+ *             - \tilde{\lambda}_{v^\Psi}^\mathrm{int}\gamma_2
+ *             v^{\mathrm{int},\Psi} , \\
+ *   D_{\Phi_{i}}
+ *              &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},+}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},-}\right)n_i^{\mathrm{ext}}
+ *                + \tilde{\lambda}_{v^0}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},0}_{i}
+ *                \notag \\
+ *              &-
+ *                \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *                v^{\mathrm{int},+}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *                v^{\mathrm{int},-}\right)n_i^{\mathrm{int}}
+ *                - \tilde{\lambda}_{v^0}^{\mathrm{int}}
+ *                v^{\mathrm{int},0}_{i},
+ * \f}
+ *
+ * with characteristic fields
+ *
+ * \f{align}{
+ *   v^{\Psi} &= \Psi, \\
+ *   v^{0}_{i} &= (\delta^k_i-n^k n_i)\Phi_{k}, \\
+ *   v^{\pm} &= \Pi\pm n^i\Phi_{i} -\gamma_2 \Psi,
+ * \f}
+ *
+ * and characteristic speeds
+ *
+ * \f{align}{
+ *   \lambda_{v^\Psi} =& -v^i_g n_i, \\
+ *   \lambda_{v^0} =& -v^i_g n_i, \\
+ *   \lambda_{v^\pm} =& \pm 1 - v^i_g n_i,
+ * \f}
+ *
+ * where \f$v_g^i\f$ is the mesh velocity. We have also defined
+ *
+ * \f{align}{
+ *   \tilde{\lambda}_{\hat{\alpha}} =
+ *    \left\{
+ *        \begin{array}{ll}
+ *          \lambda_{\hat{\alpha}} &
+ *            \mathrm{if}\;\lambda_{\hat{\alpha}}\le 0.0 \\
+ *          0 & \mathrm{otherwise}
+ *        \end{array}\right.
+ * \f}
+ *
+ * Note that we have assumed \f$n_i^{\mathrm{ext}}\f$ points in the same
+ * direction as \f$n_i^{\mathrm{int}}\f$. If \f$n_i^{\mathrm{ext}}\f$ points in
+ * the opposite direction the external speeds have their sign flipped and the
+ * \f$\pm\f$ fields and their speeds reverse roles. Specifically, in the code we
+ * have:
+ *
+ * \f{align}{
+ *   D_{\Psi} &= \bar{\lambda}_{v^{\Psi}}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},\Psi}
+ *                - \tilde{\lambda}_{v^{\Psi}}^{\mathrm{int}}
+ *                v^{\mathrm{int},g}, \\
```

Comment:
this has a `g` in the superscript, which I think must be a mistake

Reply:
Yea, that's a typo left over from the GH upwind penalty... Thanks :)